### ISSUE TITLE
[FL] Information displayed for VA health should not change until new search is conducted

### ISSUE THREAD

**Issue by randimays** [department-of-veterans-affairs/va.gov-cms#20370]
## Description

In the Facility Locator, when search results are shown for `VA health` facilities, there is a health connect phone number shown under each search result. When you click the `Facility type` dropdown and change the type to something else, such as `Urgent care`, the health connect number disappears. This is likely because the health connect number shows when the facility type is "VA health," and there is logic behind the scenes changing this value before a new search is conducted. We should wait until a new search is conducted to change the display of information for search results.

https://github.com/user-attachments/assets/a362bc26-346f-4efc-b8cf-81672abac281

Example page: https://www.va.gov/find-locations/?page=1&address=78258&facilityType=health&serviceType&latitude=29.6267&longitude=-98.495114&radius=10&bounds%5B%5D=-99.245114&bounds%5B%5D=28.8767&bounds%5B%5D=-97.745114&bounds%5B%5D=30.3767&context=San%20Antonio%2C%20Texas%2078258%2C%20United%20States

The same issue will exist for VAMC services autosuggest. When you are looking at search results from a recent search, and you change the **Service type**, Redux is updated and thus the "Showing X of X results..." text is updated live even though a new search has not been conducted.

### User story
> _Describe the audience/user, enhancement or fix, and value / outcome desired._

**AS A**
**I WANT**
**SO THAT**

### Engineering notes / background

This is related to a deeper issue with using Redux to track live updates to the search inputs. The elements below the search form ("Showing X of X results..." text, results themselves) are plugged into Redux to get search results data, but the search form is actively updating Redux at the same time.

We should be using **local component state** to manage the values of the search form inputs and only update Redux when the search is actually conducted (search button clicked, "enter" key typed, etc.). That way, Redux will represent a snapshot of the previous search data that does not change when the search form is being updated.

### Scope / Impact analysis
_What, if anything, could break as a result of this change?_
_Engineer should assess this when approaching PR._

### Roles / assignments
_After functional testing, code review, accessibility review, and design review can happen in parallel._
- [ ] **Functional testing:** 
- [ ] **Code review:** 
- [ ] **Design review:** 
- [ ] **Accessibility review:** 
- [ ] **Product acceptance:** 

## Acceptance criteria
_ACs should capture expected behavior, to inform test plans. Consider devices, documentation updates including KBs, change management, and [content model](https://prod.cms.va.gov/admin/structure/cm_document) when applicable._

- [ ] Ensure that the search form can only be submitted when the search form values have changed in order to reduce Mapbox queries.
- [ ] Impact analysis is filled out and QA test plan is updated accordingly
- [ ] Does this product have an existing regression test plan?
	- [ ] Yes, link: 
	- [ ] No, regression test plan will be created as part of the work / points in this ticket

## Definition of done
- [x] Code completed; PR created
- [x] Review instance created/available; internal team notified to start reviews
- [ ] Automated tests passed 
- [ ] Code review (meets established quality and security standards)
- [ ] PR approved
- [ ] Design review (meets design & UX requirements)
- [ ] A11y review (meets accessibility standards)
- [x] Functional/feature review (meets acceptance criteria)
- [ ] Testing completed; dev team notified to deploy
- [ ] PO/PM/external verify in prod
- [ ] Documentation updated

**Comment by jilladams** [comment]
Noted during scrum: there are 5 cases defined that display a different set of facility metadata in the results list (the rest are standard and behave the same): 
* COVID 19
* Pharmacy
* Urgent care
* Emergency
* Community care provider

**Comment by mmiddaugh** [comment]
this is a great catch - thank you! 
The health connect number should be displayed for VA health locations, regardless of the search. 
My assumption is that the same behavior is seen when facility type = VA emergency care. If so - let's close that gap too

**Comment by ACParker89** [comment]
> [@r-bartlett-gsa](https://github.com/r-bartlett-gsa) I have sorted out the issues you ran into before and have a new RI available for testing [here](http://0aaaa29bd69ba35dfd3aca5966abf40d.vfs.va.gov/)

The review instance was removed prior to review was complete, and a new one has not been able to be created since. 

@bryan-thompsoncodes @r-bartlett-gsa @JonathanKatora @mmiddaugh

**Comment by bryan-thompsoncodes** [comment]
@r-bartlett-gsa this fix is now ready for QA review [here](http://d214804d8ebfdfb60041c5cd66e0fdf1.vfs.va.gov/)

**Comment by bryan-thompsoncodes** [comment]
During code review there was an [issue noticed](https://dsva.slack.com/archives/C0FQSS30V/p1768246896198749?thread_ts=1768235876.493629&cid=C0FQSS30V) when attempting to load the form with a URL containing existing params. The search still occurs correctly but the results header and search form do not sync up due race conditions that React creates when preparing the forms components. I've been looking for a way to address this today but haven't figured this out quite yet.

**Comment by bryan-thompsoncodes** [comment]
This work is ready for review but running into review instances failing, support request opened [here](https://dsva.slack.com/archives/CBU0KDSB1/p1769011968225839)
